be/btree.c

```diff
@@ -13159,6 +14174,287 @@ static void ut_btree_crc_persist_test_internal(struct m0_btree_type   *bt,
 	btree_ut_fini();
 }
 
+static void ut_mtree_mthread_test(void)
+{
+	int                          rc;
+	int 			     ksize;
+	int 			     vsize;
+	struct btree_ut_thread_info *ti;
+	int                          i;
+	int                          k;
+	int                          cur_arr_id = 0;
+	uint16_t                     cpu;
+	void                        *rnode;
+	struct m0_btree_op           b_op       = {};
+	struct btree_mthreads_data {
+		struct m0_btree_type   btree_type;
+		uint64_t               node_size;
+		uint64_t               no_of_trees;
+	} btrees_mthreads[] = {
+		{
+			{
+				M0_BT_BALLOC_GROUP_EXTENTS,
+				8,
+				8
+			},
+                        4096, // BALLOC_ROOT_NODE_SIZE
+			      // from balloc.h
+			1
+		},
+		{
+			{
+				M0_BT_BALLOC_GROUP_DESC,
+				8,
+				8
```

Comment:
Are you sure the group descriptor is having key and value of 8 bytes each ???
Please make sure to have the data structures here of the same size as what they are in the Motr code.

Reply:
Yes... at least I see that from code 
```C
3192         bt = (struct m0_btree_type){.tt_id = M0_BT_BALLOC_GROUP_DESC,
3193                 .ksize = M0_MEMBER_SIZE(struct m0_balloc_group_desc,
3194                                         bgd_groupno),
3196         };
```
where `bgd_groupno` is `uint64_t` 
```C
 55         m0_bindex_t             bgd_groupno;
 ```
and 
```C 
 80 typedef uint64_t m0_bindex_t;
 ```

Reply:
Check again please - the value contains m0_balloc_group_desc structure you can see it in the routine balloc_trees_create().
Also confirm the sizes of the other Keys/Values in the rest of the trees as I have not done it.

Reply:
Done